### ISSUE TITLE
A Guide to Scikit-learn Pipeline

### ISSUE THREAD

**Issue by rhiriyappa** [rhiriyappa/data-science-articles#8]
##TL;DR

ML needs to execute a sequence of typical tasks:
data normalization, imputation of missing values, outlier elicitation, dimensionality reduction, classification. 

Scikit-learn's Pipeline class is designed as a manageable way to apply a series of data transformations followed by the application of an estimator. In fact, that's really all it is:

Pipeline of transforms with a final estimator.

The following picture illustrates both the training and testing data flow within the pipeline structure (copyright by Sebastian Raschka):

![image](https://user-images.githubusercontent.com/15959961/113774123-b7c3d180-96db-11eb-9dae-16ccef8a4c56.png)

##Article Link

https://medium.com/vickdata/a-simple-guide-to-scikit-learn-pipelines-4ac0d974bdcf
https://iaml.it/blog/optimizing-sklearn-pipelines

##Key Takeaway

Pipeline is useful for:

1. Convenience in creating a coherent and easy-to-understand workflow
2. Enforcing workflow implementation and the desired order of step applications
3. Reproducibility
4. Value in persistence of entire pipeline objects (goes to reproducibility and convenience)
